mautic/mautic
# Remove deprecated mauticfactory dependency from mailhelper.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [X ]
| Deprecations?                          | [ getMailer and getSampleMailer from Mailhelper, to be replaced with default mailHelper reset function.]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

MauticFactory is deprecated from M2 and scheduled for removal in M3. That did not happen, but let's hope we can remove it in M6. To do this, all internal usage of it must be gone by the time we get to M6. But removal of internal usage can happen in M5. So let's do it. 
This removes the usage from the MailHelper class, which was a big user. 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test all functionality MailHelper touches, or trust on testing and phpstan.
